File: src/main/java/ru/dargen/evoplus/EvoPlus.java
package ru.dargen.evoplus;

import com.google.gson.Gson;
import com.google.gson.GsonBuilder;
import lombok.Getter;
import lombok.experimental.Accessors;
import net.fabricmc.api.ModInitializer;
import net.fabricmc.loader.FabricLoader;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;
import ru.dargen.evoplus.command.CommandManager;
import ru.dargen.evoplus.command.commands.HelpCommand;
import ru.dargen.evoplus.command.commands.TeamCommand;
import ru.dargen.evoplus.command.commands.WarCommand;
import ru.dargen.evoplus.event.EventBus;
import ru.dargen.evoplus.feature.Feature;
import ru.dargen.evoplus.feature.FeatureManager;
import ru.dargen.evoplus.feature.impl.MiscFeature;
import ru.dargen.evoplus.feature.impl.RenderFeature;
import ru.dargen.evoplus.feature.impl.TeamWarFeature;
import ru.dargen.evoplus.feature.impl.boss.BossTimerFeature;
import ru.dargen.evoplus.feature.impl.staff.StaffTimerFeature;
import ru.dargen.evoplus.feature.impl.stats.StatsFeature;
import ru.dargen.evoplus.notify.NotificationsManager;
import ru.dargen.evoplus.task.TaskBus;
import ru.dargen.evoplus.util.diamondworld.EvoStatistic;

import java.io.File;

@Getter
public class EvoPlus implements ModInitializer {

    public final static String NAME = "§eEvo§6Plus";
    public final static String PREFIX = "§7[" + NAME + "§7] §r";

    @Getter
    @Accessors(fluent = true)
    protected static EvoPlus instance;

    @Getter
    @Accessors(fluent = true)
    public final static File folder = new File(System.getProperty("user.dir"), "evo-plus");
    @Getter
    @Accessors(fluent = true)
    private static final String version = FabricLoader.INSTANCE
            .getModContainer("evo-plus")
            .get()
            .getMetadata()
            .getVersion()
            .getFriendlyString();
    private final Gson prettyGson = new GsonBuilder().setPrettyPrinting().create();
    private final Gson gson = new GsonBuilder().create();
    private final Logger logger = LogManager.getLogger("EvoPlus");

    private TaskBus taskBus;
    private EventBus eventBus;
    private EvoStatistic stats;
    private FeatureManager featureManager;
    private CommandManager commandManager;
    private NotificationsManager notifyManager;

    @Override
    public void onInitialize() {
        instance = this;
        taskBus = new TaskBus(this);
        eventBus = new EventBus(this);
        stats = new EvoStatistic(this);
        featureManager = new FeatureManager(this);
        commandManager = new CommandManager(this);
        notifyManager = new NotificationsManager(this);

        commandManager.registerCommands(
                new TeamCommand(), new WarCommand(), new HelpCommand()/*, new ClanWebHookCommand()*//*, new SellCommand()*/
        );

        Feature.STATS_FEATURE = new StatsFeature();
        Feature.BOSS_TIMER_FEATURE = new BossTimerFeature();
        Feature.STAFF_TIMER_FEATURE = new StaffTimerFeature();
        Feature.TEAM_WAR_FEATURE = new TeamWarFeature();
//        Feature.CLAN_FEATURE = new ClanFeature();
        Feature.RENDER_FEATURE = new RenderFeature();
        Feature.MISC_FEATURE = new MiscFeature();
    }

}

File: src/main/kotlin/ru/dargen/evoplus/EvoPlus.kt
package ru.dargen.evoplus

import net.fabricmc.api.ClientModInitializer
import net.fabricmc.loader.api.FabricLoader
import ru.dargen.evoplus.api.event.EventBus
import ru.dargen.evoplus.api.keybind.KeyBindings
import ru.dargen.evoplus.api.render.animation.AnimationRunner
import ru.dargen.evoplus.api.render.context.Overlay
import ru.dargen.evoplus.api.render.context.World
import ru.dargen.evoplus.api.render.node.leftClick
import ru.dargen.evoplus.api.scheduler.Scheduler
import ru.dargen.evoplus.api.scheduler.scheduleEvery
import ru.dargen.evoplus.feature.Features
import ru.dargen.evoplus.features.misc.Notifies
import ru.dargen.evoplus.protocol.EvoPlusProtocol
import ru.dargen.evoplus.util.Updater
import ru.dargen.evoplus.util.minecraft.Client
import java.util.concurrent.TimeUnit


val ModLabel = "§f§lEvo§6§lPlus"

val Logger = java.util.logging.Logger.getLogger("EvoPlus")

object EvoPlus : ClientModInitializer {

    val ModContainer by lazy { FabricLoader.getInstance().getModContainer("evo-plus").get() }
    
    override fun onInitializeClient() {
        EventBus
        Scheduler
        KeyBindings
        ReplacerParser

        EvoPlusProtocol

        World
        Overlay
        AnimationRunner

        Features

        scheduleUpdater()
    }

    private fun scheduleUpdater() = scheduleEvery(5, 5, unit = TimeUnit.MINUTES) {
        if (Client?.inGameHud != null && Updater.Outdated) Notifies.showText(
            "Обнаружена новая версия EvoPlus - §e${Updater.LatestVersion}.",
            "Нажмите, чтобы обновиться.",
            delay = 15.0
        ) { leftClick { _, state -> if (isHovered && state) Updater.tryUpdate() } }
    }

}